s3tests_boto3/functional/test_s3.py

```diff
@@ -3522,6 +3522,37 @@ def test_object_put_authenticated():
     response = client.put_object(Bucket=bucket_name, Key='foo', Body='foo')
     assert response['ResponseMetadata']['HTTPStatusCode'] == 200
 
+def _test_object_presigned_put_object_with_acl(client=None):
+    if client is None:
+        client = get_client()
+
+    bucket_name = get_new_bucket()
+    client.create_bucket(Bucket=bucket_name)
```

Comment:
[get_new_bucket()](https://github.com/ceph/s3-tests/blob/da91ad8bbf899c72199df35b69e9393c706aabee/s3tests_boto3/functional/__init__.py#L584-L596) actually creates the bucket using the default `get_client()`, so `client.create_bucket()` always recreates it. in the tenant test, this recreation is expected to fail:

> An error occurred (BucketAlreadyExists) when calling the CreateBucket operation: Cannot modify existing access control policy

strangely, i'm only seeing it fail that way in the rgw/dbstore suite

you can either call `get_new_bucket(client)` with the correct client, or call `get_new_bucket_name()` just to generate the name

Reply:
oh, the tenanted user would create a different bucket with the same name. i guess dbstore just doesn't handle the tenants correctly

Reply:
ah nice catch, I've changed to your suggestion ``get_new_bucket(client)``